### PR TITLE
Add CI/CD workflows for Cloud Functions

### PR DESCRIPTION
**Pull request by hernandezc1**
The goal of this PR is to develop a CI/CD workflow that tests changes to any of the Pitt-Google Broker's microservices when triggered by the following actions:
- workflow_dispatch
- push
- pull

I appreciate any feedback, as well as any references to documentation of best practices for future improvements.

### CODE REVIEW

File: broker/cloud_functions/classify_snn/tests/test_cloud_fnc.py
Comment (hernandezc1) [inline broker/cloud_functions/classify_snn/tests/test_cloud_fnc.py:1]: This (fake) test case was purposely done in order for Daniel to show me that the test would successfully run

**Comment by Crash-GHaun** [comment]
LGTM, although I'm not super familiar with GitHub Actions :)

**Review by Crash-GHaun — APPROVED** [review]

File: .github/workflows/TestCloudFunction.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
+name: Test Cloud Functions
+
+on:
+  workflow_dispatch:
+  push:
+  pull_request:
```
Comment (djperrefort) [inline .github/workflows/TestCloudFunction.yml]: You don't need to include the `pull_request` option if you already have `push` enabled. 

There are cases where both are necessary. E.g.,when triggering on pushes/PRs on specific branches. However, this is not one of those cases.

File: .github/workflows/TestCloudFunction.yml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+          coverage report
+
+      # Report test coverage to codacy for the current set of tests
+      - name: Report partial coverage results
+        run: bash <(curl -Ls https://coverage.codacy.com/get.sh) report --partial -l Python -r report_${{ matrix.python-version }}.xml
+        env:
+          CODACY_PROJECT_TOKEN: ${{ secrets.CODACY_PROJECT_TOKEN }}
+      
+      
+    codacy-coverage-reporter:
```
Comment (djperrefort) [inline .github/workflows/TestCloudFunction.yml]: `codacy-coverage-reporter` must be indented with two spaces, not four.

**Review by djperrefort — CHANGES_REQUESTED** [review]

**Review by djperrefort — APPROVED** [review]

**Pull request merged**